File: tools/debug-sections.js
const sqlite3 = require('sqlite3').verbose();
const path = require('path');

const dbPath = path.resolve(__dirname, '../assets/risale.db'); // Adjust path if needed. risaleRepo uses 'contentDb' which usually opens a local DB file.
// Wait, risaleRepo uses `getDb`. Let's assume it's `src/services/db/risale.db` or similar? 
// The error `src/services/db/risale.db` path I used before might be wrong if I guessed it.
// Let's check `src/services/contentDb.ts` to see where the DB is.

const db = new sqlite3.Database('assets/risale.db', (err) => {
    if (err) {
        console.error('Could not open DB:', err.message);
        // Try default location if any
    } else {
        console.log('Connected to DB');
        checkSection('birinci_soz');
    }
});

function checkSection(id) {
    db.serialize(() => {
        db.all("SELECT COUNT(*) as count FROM paragraphs WHERE section_id = ?", [id], (err, rows) => {
            if (err) console.error(err);
            else console.log(`Chunks for ${id}:`, rows[0].count);
        });

        db.all("SELECT DISTINCT section_id FROM paragraphs LIMIT 10", [], (err, rows) => {
            if (err) console.error(err);
            else console.log('Sample Section IDs:', rows);
        });
    });
}

File: tools/debug_sections.py
import sqlite3
import os

db_path = 'assets/risale.db'

if not os.path.exists(db_path):
    print(f"DB not found at {db_path}")
    exit(1)

conn = sqlite3.connect(db_path)
cursor = conn.cursor()

# Check count for First Word
sid1 = 'sozler-section-001'
cursor.execute("SELECT COUNT(*) FROM paragraphs WHERE section_id = ?", (sid1,))
count1 = cursor.fetchone()[0]
print(f"{sid1}: {count1} chunks")

# Check count for Second Word
sid2 = 'sozler-section-003'
cursor.execute("SELECT COUNT(*) FROM paragraphs WHERE section_id = ?", (sid2,))
count2 = cursor.fetchone()[0]
print(f"{sid2}: {count2} chunks")

# Check distinct section IDs to verify what exists
cursor.execute("SELECT DISTINCT section_id FROM paragraphs LIMIT 10")
print("Sample distinct section_ids:", cursor.fetchall())

# Check if First Word contains text 'İkinci Söz' (indicating leak)
cursor.execute("SELECT text FROM paragraphs WHERE section_id = ? AND text LIKE '%İkinci Söz%'", (sid1,))
leaks = cursor.fetchall()
if leaks:
    print(f"Possible Leak in {sid1}: Found 'İkinci Söz' inside it.")
    for l in leaks:
        print(f" - {l[0][:50]}...")
else:
    print(f"No explicit 'İkinci Söz' header found in {sid1}.")

conn.close()
